src/NUnitEngine/nunit.engine.core/Services/ExtensionManager.cs

```diff
@@ -416,18 +459,29 @@ internal void FindExtensionsInAssembly(ExtensionAssembly assembly)
             }
 #endif
 
-            foreach (var type in assembly.MainModule.GetTypes())
+            foreach (var extensionType in assembly.MainModule.GetTypes())
             {
-                CustomAttribute extensionAttr = type.GetAttribute("NUnit.Engine.Extensibility.ExtensionAttribute");
+                CustomAttribute extensionAttr = extensionType.GetAttribute("NUnit.Engine.Extensibility.ExtensionAttribute");
 
                 if (extensionAttr == null)
                     continue;
 
-                object versionArg = extensionAttr.GetNamedArgument("EngineVersion");
-                if (versionArg != null && new Version((string)versionArg) > ENGINE_VERSION)
-                    continue;
+                // TODO: This is a remnant of older code. In principle, this should be generalized
+                // to something like "HostVersion". However, this can safely remain until
+                // we separate ExtensionManager into its own assembly.
+                string versionArg = extensionAttr.GetNamedArgument("EngineVersion") as string;
+                if (versionArg != null)
+                {
+                    Assembly THIS_ASSEMBLY = Assembly.GetExecutingAssembly();
```

Comment:
This is constant for this run, but obtained for every attribute found. Could be set before searching anything.